File: Majo/Act-Instructor_Henry/lenguajes/dart/orientado_a_objeto/clases.dart
class Calculadora{
 double? num1;
 double? num2;
 String? operacion;

 Calculadora (double n1, double n2, String operacion){
    this.num1 = n1;
    this.num2 = n2;
    this.operacion = operacion;
 }
    double sumar() {
        return num1! + num2!;
    }

    double restar() {
        return num1! - num2!;  
    }

    double multiplicar() {
        return num1! * num2!;
    }

    double dividir() {
    if (num2 != 0) {
      return num1! / num2!;
    } else {
      print("Error: División por cero");
      return double.nan;
    }
}

  void calcular() {
    if (operacion == "SUMA") {
      print(sumar());
    } else if (operacion == "MULTIPLICACION") {
      print(multiplicar());
    } else if (operacion == "DIVISION") {
      print(dividir());
    } else if (operacion == "RESTA") {
      print(restar());
    } else {
      print("Operación no válida");
    }
  }
}

void main() {
  Calculadora calculo = Calculadora(10, 2, "SUMA");
  calculo.calcular();
}

File: Majo/Act-Instructor_Henry/lenguajes/ruby/orientado_a_objeto/clases.rb
class Calculadora
    attr_accessor :num1, :num2, :operacion
  
    def initialize(n1, n2, operacion)
      @num1 = n1
      @num2 = n2
      @operacion = operacion
    end
  
    def sumar
      @num1 + @num2
    end
  
    def restar
      @num1 - @num2
    end
  
    def multiplicar
      @num1 * @num2
    end
  
    def dividir
      if @num2 != 0
        @num1 / @num2
      else
        puts "Error: División por cero"
        Float::NAN
      end
    end
  
    def calcular
      case @operacion
      when "SUMA"
        puts sumar
      when "MULTIPLICACION"
        puts multiplicar
      when "DIVISION"
        puts dividir
      when "RESTA"
        puts restar
      else
        puts "Operación no válida"
      end
    end
  end
  
  def main
    calculo = Calculadora.new(10, 2, "SUMA")
    calculo.calcular
  end
  
  main
  